### DISCUSSION TITLE
Please help on "MAP rule invalid" error.

### DISCUSSION THREAD

**Discussion by arihid** [fakemanhk/openwrt-jp-ipoe#21] [Q&A]
Hi!

I have followed through the steps to connect the router, but I always had "MAP rule invalid" error.
I have set the ip6 prefix to 240b:aaaa:bbbb:cccc::/64 (in my case, ip6 is usually starts in 240b), and tried to redo the whole setup, but it isn't helping.

Is there any other possible solution? For now I simply use my openwrt as an AP hooked to the provided router.

I'm using Buffalo WXR-5950AX12 running OpenWRT 23.05.00-rc3, my network provider is HOME光.

**Comment by fakemanhk** [comment]
Are you sure your provider is using MAP-E? If your Buffalo 5950AX12 stock firmware is able to work with your provider, try to look into the status page to see if it is really using MAP-E.

**Comment by arihid** [comment]
Oh yeah, I wonder why I never asked that question...
Tried to connect in WXR-5950AX stock firmware once, and I can't connect at all...
In my attempt on OpenWRT, I can connect to the provider using DHCPv6 Client, I got the IP successfully, but the MAP tunnel always says "MAP rule invalid"
Here's the status page in the provided router:
![IMG_20230903_234639](https://github.com/fakemanhk/openwrt-jp-ipoe/assets/82217590/f97f314f-ccd7-4bdf-94f0-5d48b25e269a)
I don't know if I can try pppoe, since I don't receive any credential information from my provider.

**Reply by arihid** [reply]
Oh yeah, I forgot to mention that I didn't subscribe to Hikari Denwa service.

**Comment by fakemanhk** [comment]
> Oh yeah, I wonder why I never asked that question... Tried to connect in WXR-5950AX stock firmware once, and I can't connect at all... In my attempt on OpenWRT, I can connect to the provider using DHCPv6 Client, I got the IP successfully, but the MAP tunnel always says "MAP rule invalid" Here's the status page in the provided router: ![IMG_20230903_234639](https://user-images.githubusercontent.com/82217590/265271632-f97f314f-ccd7-4bdf-94f0-5d48b25e269a.jpg) I don't know if I can try pppoe, since I don't receive any credential information from my provider.

v6プラス, so you should be using MAP-E as well. When you use the Buffalo router with stock firmware, the status page should tell you some CE address and some other information that should be the same as the calculator, is it? If not then something is wrong there.

For 10G line there should be no any PPPoE AFAIK.

**Comment by arihid** [comment]
I finally can connect to the map-e network. After I have some free time to tinker things again, I found out that it is my configuration that is faulty. After carefully reviewing the tutorial and the config I made, I am now able to reach the ip4 network.

I will close this discussion. Thank you for your reply.

**Reply by LIII-XXII** [reply]
If I may, have you found out what the issue was?

**Reply by arihid** [reply]
I put a wrong ip6prefix value into wan6 interface.

It is mentioned in the README.MD that you can't access the tunnel without correct ip6prefix.

**Comment by fakemanhk** [comment]
> I put a wrong ip6prefix value into wan6 interface.

Yes, I was actually struggling on this issue for a few months, had given up some time and when I come back to the same problem, suddenly found one Japanese website using Linux with the manual prefix setup which made me try to set this value and success.
